igvm_c/Makefile

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: MIT OR Apache-2.0
```

Comment:
I wonder if we could also/instead have a rust sample using the C api? Might be nicer to integrate with cargo test, as well?

Reply:
Or maybe just some rust tests for the C API would be good enough?

Reply:
I'm inclined to push Rust users to the Rust API, so I think no rust sample is fine. That way no bad examples in the codebase.